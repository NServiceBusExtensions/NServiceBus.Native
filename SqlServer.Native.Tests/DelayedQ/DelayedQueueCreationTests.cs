using ApprovalTests;
using NServiceBus.Transport.SqlServerNative;
using Xunit;

public class DelayedQueueCreationTests
{
    [Fact]
    public void Run()
    {
        DbSetup.Setup();
        using (var connection = Connection.OpenConnection())
        {
            var manager = new DelayedQueueManager("DelayedQueueCreationTests", connection);
            manager.Drop().Await();
            manager.Create().Await();
            var sqlScriptBuilder = new SqlScriptBuilder(tables:true, namesToInclude: "DelayedQueueCreationTests");
            Approvals.Verify(sqlScriptBuilder.BuildScript(connection));
        }
    }
}