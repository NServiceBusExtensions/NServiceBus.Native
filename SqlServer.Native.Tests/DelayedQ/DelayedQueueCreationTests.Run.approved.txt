CREATE TABLE [dbo].[DelayedQueueCreationTests](
	[Headers] [nvarchar](max) COLLATE Latin1_General_CI_AS NOT NULL,
	[BodyString]  AS (CONVERT([varchar](max),[Body])),
	[Body] [varbinary](max) NULL,
	[Due] [datetime] NOT NULL,
	[RowVersion] [bigint] IDENTITY(1,1) NOT NULL
) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
